src/intersection-observer.js

```diff
@@ -218,10 +218,14 @@ export class IntersectionObserver extends Observable {
   }
 
   /**
-   * Provice a function to clear timeout before set this intersection to null.
+   * Provide a function to clear timeout before set this intersection to null.
    */
   destroy() {
     timer.cancel(this.flushTimeout_);
     this.flushTimeout_ = 0;
+    if (this.unlistenViewportChanges_) {
+      this.unlistenViewportChanges_();
+      this.unlistenViewportChanges_ = null;
+    }
```

Comment:
I didn't call `onViewportCallback(false)` because I don't want to call `fire()` in `destroy()` function, but I did add the double tap that check `onScroll` and `onChange` listener are unlistened. 
Nice Video BTW!
@dvoytenko